cmd/root_test.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"os"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+	exitCode := m.Run()
+
+	os.Exit(exitCode)
+}
+
+// Specify arguments and the expected outputs of rootCmd on them
+var errorOutput = readFile("nonexistent") + "\n"
+var goModOutput = readFile("../go.mod")
+var inputData = "Your test input data here\n"
+var flagtests = []struct {
+	inputArgs []string
+	output    string
+}{
+	{[]string{}, inputData},                                           //Tests no args -> Stdin
+	{[]string{"-"}, inputData},                                        //Tests "-" -> Stdin
+	{[]string{"../go.mod"}, goModOutput},                              //Tests reading files
+	{[]string{"nonexistent"}, errorOutput},                            //Tests reporting file open errors without exiting
+	{[]string{"../go.mod", "-"}, goModOutput + inputData},             //Tests file to Stdin transition
+	{[]string{"-", "../go.mod"}, inputData + goModOutput},             //Tests Stdin to file transition
+	{[]string{"-", "nonexistent"}, inputData + errorOutput},           //Tests Stdin to error transition
+	{[]string{"nonexistent", "-"}, errorOutput + inputData},           //Tests error to Stdin transition
+	{[]string{"../go.mod", "nonexistent"}, goModOutput + errorOutput}, //Tests file to error transition
+	{[]string{"nonexistent", "../go.mod"}, errorOutput + goModOutput}, //Tests error to file transition
+}
+
+func TestFlagArgs(t *testing.T) {
+	var output []byte
+	var bytes []byte
+
+	// Create temporary files to read/write Stdin and Stdout
+	stdin, errIn := os.CreateTemp("", "testStdin")
+	handleErr(errIn)
+	defer os.Remove(stdin.Name())
+	stdout, errOut := os.CreateTemp("", "testStdout")
+	handleErr(errOut)
+	defer os.Remove(stdout.Name())
+
+	os.Stdin = stdin
+	os.Stdout = stdout
+	stdin.Write([]byte(inputData))
+
+	for i, flag := range flagtests {
+		// Reset initial conditions between each test
+		err := os.Truncate(os.Stdout.Name(), 0)
+		errHandle(err)
+		os.Stdin.Seek(0, 0)
+
+		rootCmd.SetArgs(flag.inputArgs)
+		rootCmd.Execute()
+		os.Stdout.Seek(0, 0)
+
+		bytesRead, err := io.ReadAll(os.Stdout)
+		handleErr(err)
```

Comment:
You have two functions that do basically the same thing, named `errHandle` and `handleErr`. Remember that in Go you do not have to explicitly import members from the same package, they are compiled together all the same. You have access to `errHandle` in this file and `handleErr` in the main file.

That said, it is generally considered bad form in Go to have a function that only does `if err != nil` for you. Each error should, ideally, be handled uniquely or not at all. You may want to set up some autocomplete snippets for standard patterns, but that's up to you to configure your editor.